src/components/quick-settings.tsx

```diff
@@ -0,0 +1,52 @@
+import { TranslationBundle } from '@jupyterlab/translation';
+import type { CommandRegistry } from '@lumino/commands';
+import { settingsIcon } from '@jupyterlab/ui-components';
+import * as React from 'react';
+import { Menu } from '@lumino/widgets';
+
+import { CommandIDs } from '../types';
+
+export function QuickSettings(props: {
+  trans: TranslationBundle;
+  commands: CommandRegistry;
+}): React.ReactElement {
+  const { commands } = props;
+
+  const menu = new Menu({ commands: commands });
+  menu.addItem({ command: CommandIDs.showStarred, args: {} });
+  menu.addItem({ command: CommandIDs.searchAllSections, args: {} });
+  menu.addItem({
+    command: 'settingeditor:open',
+    args: { query: 'New Launcher' }
+  });
+
+  const iconRef = React.useRef<HTMLDivElement>(null);
+
+  const onClickHandler = (event: React.MouseEvent) => {
+    const current = iconRef.current;
+    let x, y;
+    if (current) {
+      const position = current.getBoundingClientRect();
+      x = position.left;
+      y = position.bottom;
+    } else {
+      x = event.clientX;
+      y = event.clientY;
+    }
+    menu.open(x, y);
```

Comment:
Currently the menu is positioned within the browser window:

![image](https://github.com/nebari-dev/jupyterlab-new-launcher/assets/5832902/5fb03b26-6ad1-438f-90c0-450aa6bdc5ba)

I wonder if it would be nicer if it was positioned relative to the launcher body or the button. You can pass a third argument to `menu.open()`  with [`Menu.IOpenOptions`](https://lumino.readthedocs.io/en/latest/api/interfaces/widgets.Menu.IOpenOptions.html) object. Once https://github.com/jupyterlab/lumino/pull/700 is released we will be able to pass a custom `host` but for now we could only try using `forceX` to improve the positioning.

Reply:
It currently should be positioned to the button. It's just shifting the menu so it fits in the window. This is what it looks like when the panel isn't directly next the edge of the window:

![image](https://github.com/nebari-dev/jupyterlab-new-launcher/assets/13774419/51e7e3bb-670d-42fc-933a-e888b6a5bbdd)

If I use `forceX` the part of the menu isn't viewable which I think may be worse:

![image](https://github.com/nebari-dev/jupyterlab-new-launcher/assets/13774419/61f95aa7-caa0-46f8-b60f-c305641dc60d)

I wonder if there is a way to align it so the top-right corner of the menu aligns with the bottom right corner of the button. In my opinion that would look the cleanest.

Reply:
> align it so the top-right corner of the menu aligns with the bottom right corner of the button. In my opinion that would look the cleanest.

:100: agree. There is no way as of today. Can you open a feature request on lumino, and another issue here to track it as a follow-up enhancement?

Reply:
Sounds good. I opened [Lumino#709](https://github.com/jupyterlab/lumino/issues/709) for this